string color
float32 distance
